paper.md

```diff
@@ -2,11 +2,77 @@
+Docker has been the de facto container runtime, with other container runtimes such as Singularity [@kurtzer_singularity_2017], Rkt [@rktcommunity_rkt_2019], Charliecloud [@priedhorsky_charliecloud_2017] and Podman [@podmancommunity_containers_2019] having emerged. 
+The Linux Foundation bootstrapped the Open Container Initiative (OCI) [@opencontainerinitiative_new_2016] and is close to releasing version 1.0 of a container image and runtime specifications. 
+With Docker, began the era of container-native software development, which is a paradigm that promotes the building, testing, and deployment of software in containers. 
+Since, these container runtimes are available for almost every well known operating systems and architectures, experiments can be reproduced easily using containerized workflows in almost any environment [@stubbs2016endofday] [@zheng_integrating_2015].
+
+<!-- what problem is remaining and what are the contributions -->
+
+Although, there are different container engines available, switching between them is hard as they have different API's and image formats and also due to the absence of tools that allow running containerized workflows in an engine agnostic way.
+It has also been found that as scientific workflows become increasingly complex, continuous validation of the workflows which is critical to ensuring good reproducibility, becomes difficult [@deelman2018future] [@cohen2017scientific].
+Currently, the different container based workflow engines that are available involves engines that are inherently container-native but assume the presence of a fully provisioned Kubernetes [@kubernetes_google] cluster at their disposal for workflow execution. 
+Therefore, this category of workflow engines could also be termed as cloud-native [@balalaie2016microservices]. 
+Some popular examples of this type of workflow engines are Argo [@argocommunity_argoproj_2019], Pachyderm [@novella_containerbased_2018] and Brigade [@brigade]. 
+The presence of a Kubernetes cluster or a cloud computing environment should not be a hardcore requirement for reproducing any experiment in a container-native manner, since it is often costly [@rodriguez2020container] to get access to one and this inturn makes reproducibility complex. 
+Workflow engines should provide flexibility for their users to run workflows in a wide range of computing environments like a single-node local machine, a HPC environment [@yang2005high], a Kubernetes cluster or any other computing environment of their choice. 
+
+Popper [@systemslabpopper] is a light-weight workflow execution engine that follows a container-native strategy for building reproducible workflows from archived experimental artifacts. 
```

Comment:
"execution engine that allows users to follow the container-native paradigm for building..."